save-demo/src/main/kotlin/com/saveourtool/save/demo/runners/cli/CliRunner.kt

```diff
@@ -0,0 +1,88 @@
+package com.saveourtool.save.demo.runners.cli
+
+import com.saveourtool.save.demo.DemoAdditionalParams
+import com.saveourtool.save.demo.DemoResult
+import com.saveourtool.save.demo.runners.Runner
+import com.saveourtool.save.demo.utils.generateName
+import org.springframework.stereotype.Component
+import java.io.File
+import java.nio.file.Path
+import kotlin.io.path.div
+
+/**
+ * Interface that should be implemented by all the runners that use [ProcessBuilder] in order to run tools for demo.
+ */
+@Component
+interface CliRunner <in P : DemoAdditionalParams, out R : DemoResult> : Runner<P, R> {
+    /**
+     * @return string that starts with Linux, Windows or macOS
+     */
+    fun osName(): String = System.getProperty("os.name")
+
+    /**
+     * @param workingDir the directory where the tool is run
+     * @param params additional params of type [DemoAdditionalParams]
+     * @return executable file (diktat or ktlint)
+     */
+    fun getExecutable(workingDir: Path, params: P): File
```

Comment:
Avoid `java.io.File`, try to stick to NIO.2 API:

```suggestion
    fun getExecutable(workingDir: Path, params: P): Path
```

Reply:
Already changed it